tools/export/gnuarmeclipse/__init__.py

```diff
@@ -0,0 +1,345 @@
+                                      for s in source_folders)]
+        # print 'top folders'
+        # print top_folders
+
+        self.source_tree = {}
+        for top_folder in top_folders:
+            for root, dirs, files in os.walk(top_folder):
+                if len(dirs) == 0:
+                    self.add_source_folder_to_tree(root)
+
+        for folder in source_folders:
+            self.add_source_folder_to_tree(folder, True)
+
+        # print
+        # print self.source_tree
+        # self.dump_paths(self.source_tree)
+
+        # print 'excludings'
+        self.excluded_folders = []
+        self.recurse_excludings(self.source_tree)
+
+    def generate(self):
+        """
+        Override the parent method with code generation.
+        """
+        if not self.resources.linker_script:
+            raise NotSupportedException("No linker script found.")
+
+        # TODO: Check if this is needed.
+        # self.resources.win_to_unix()
+
+        libraries = []
+        print self.resources.libraries
+        for lib in self.resources.libraries:
+            l, _ = splitext(basename(lib))
+            libraries.append(l[3:])
+
+        MCPUS = {
+            'Cortex-M0': {'mcpu': 'cortex-m0', 'fpu_unit': None},
+            'Cortex-M0+': {'mcpu': 'cortex-m0plus', 'fpu_unit': None},
+            'Cortex-M1': {'mcpu': 'cortex-m1', 'fpu_unit': None},
+            'Cortex-M3': {'mcpu': 'cortex-m3', 'fpu_unit': None},
+            'Cortex-M4': {'mcpu': 'cortex-m4', 'fpu_unit': None},
+            'Cortex-M4F': {'mcpu': 'cortex-m4', 'fpu_unit': 'fpv4spd16'},
+            'Cortex-M7': {'mcpu': 'cortex-m7', 'fpu_unit': None},
+            'Cortex-M7F': {'mcpu': 'cortex-m7', 'fpu_unit': 'fpv4spd16'},
+            'Cortex-M7FD': {'mcpu': 'cortex-m7', 'fpu_unit': 'fpv5d16'},
+        }
+
+        core = self.toolchain.target.core
+
+        try:
+            # cortex-m0, cortex-m0-small-multiply, cortex-m0plus,
+            # cortex-m0plus-small-multiply, cortex-m1, cortex-m1-small-multiply,
+            # cortex-m3, cortex-m4, cortex-m7.
+            target_mcpu = MCPUS[core]['mcpu']
+
+            # default, fpv4spd16, fpv5d16, fpv5spd16
+            target_fpu_unit = MCPUS[core]['fpu_unit']
+
+            # soft, softfp, hard.
+            target_fpu_abi = 'soft'
+
+            if target_fpu_unit != None:
+                target_fpu_abi = 'hard'
+
+        except AttributeError:
+            # TODO filter out projects with toolchain core not supported,
+            # instead of raising an exception.
+            raise NotSupportedException(
+                'Target core {0} not supported.'.format(core))
+
+        # TODO: get the list from existing .cproject
+        build_folders = ['Debug', 'Release']
+
+        objects = [self.filter_dot(s) for s in self.resources.objects]
+        for bf in build_folders:
+            objects = [o for o in objects if not o.startswith(bf + '/')]
```

Comment:
Same here. Can you point to where you have normalized source paths? The resource object will contain the native os file separators.